### PR TITLE
enhance: plugin Zipkin add service name and report local server IP

### PR DESCRIPTION
**Pull request by lilien1010**
### Summary

allow zipkin to set servie name and local ip sddress so that we can track it in APM system

### Full changelog

* apisix/plugins/zipkin.lua
* doc/plugins/zipkin-cn.md
* doc/plugins/zipkin.md
* apisix/plugins/zipkin/reporter.lua
* t/lib/server.lua
* t/plugin/zipkin.t

### CODE REVIEW

File: lua/apisix/plugins/zipkin.lua [hunk 1]
```diff
@@ -42,6 +47,20 @@ local _M = {
...
+    local t, err = io.popen(cmd)
+    if not t then
+        return nil, "failed to execute command: " .. cmd .. ", error info:" .. err
+    end
+    local data = t:read("*all")
+    t:close()
+    return data
+end
+
+local LOCAL_SERVER_IP,_ = excute_cmd("hostname -I | awk '{print $1}'")
```
Comment (moonming) [inline lua/apisix/plugins/zipkin.lua]: how about nginx var `server_addr`?

File: lua/apisix/plugins/zipkin.lua [hunk 2]
```diff
@@ -109,6 +130,10 @@ function _M.rewrite(conf, ctx)
     local request_span = ctx.opentracing.request_span
     ctx.opentracing.rewrite_span = request_span:start_child_span(
                                             "apisix.rewrite", start_timestamp)
+
+    ctx.opentracing.rewrite_span:set_tag("peer.service",conf.service_name)
```
Comment (moonming) [inline lua/apisix/plugins/zipkin.lua]: add space after comma

File: t/plugin/zipkin.t [hunk 3]
```diff
@@ -126,7 +126,8 @@ done
                         "plugins": {
                             "zipkin": {
                                 "endpoint": "http://127.0.0.1:1982/mock_zipkin",
-                                "sample_ratio": 1
+                                "sample_ratio": 1,
+                                "service_name":"APISIX"
```
Comment (moonming) [inline t/plugin/zipkin.t]: Add space after `:`

File: lua/apisix/plugins/zipkin.lua (hunk 1 above)
Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua]: changed into ctx.server_addr

File: t/plugin/zipkin.t (hunk 3 above)
Comment (lilien1010) [reply t/plugin/zipkin.t]: done

File: lua/apisix/plugins/zipkin.lua (hunk 2 above)
Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua]: done

File: doc/plugins/zipkin.md [hunk 4]
```diff
@@ -39,6 +39,7 @@ It's also works with `Apache SkyWalking`, which is support Zipkin v1/v2 format.
 |---------     |--------|-----------|
 | endpoint     |required|the http endpoint of Ziplin, for example: `http://127.0.0.1:9411/api/v2/spans`.|
 | sample_ratio |required|the ratio of sample, the minimum is 0.00001, the maximum is 1.|
+| service_name |optional|service name for zipkin reporter.|
```
[the remark this answers is not in the document]
Comment (lilien1010) [reply doc/plugins/zipkin.md]: done

File: doc/plugins/zipkin-cn.md [hunk 5]
```diff
@@ -34,8 +34,9 @@
 
 ## 属性
 
-* `endpoint`: Ziplin 的 http 节点，例如`http://127.0.0.1:9411/api/v2/spans`。
+* `endpoint`: Zipkin 的 http 节点，例如`http://127.0.0.1:9411/api/v2/spans`。
 * `sample_ratio`: 监听的比例，最小为0.00001，最大为1。
+* `service_name`: 可选参数，标记当前服务的名称，默认是APISIX。
```
[the remark this answers is not in the document]
Comment (lilien1010) [reply doc/plugins/zipkin-cn.md]: change all to `APISIX`

File: lua/apisix/plugins/zipkin.lua [hunk 6]
```diff
@@ -29,7 +29,12 @@ local schema = {
     type = "object",
     properties = {
         endpoint = {type = "string"},
-        sample_ratio = {type = "number", minimum = 0.00001, maximum = 1}
+        sample_ratio = {type = "number", minimum = 0.00001, maximum = 1},
+        service_name = {
+            type = "string",
+            description = "service name for zipkin reporter",
+            default = {"APISIX"},
```
Comment (moonming) [inline lua/apisix/plugins/zipkin.lua]: should not has `{` and `}`, right?

File: lua/apisix/plugins/zipkin.lua [hunk 7]
```diff
@@ -94,6 +99,8 @@ function _M.rewrite(conf, ctx)
              -- TODO: support ipv6
             ["peer.ipv4"] = core.request.get_remote_client_ip(ctx),
             ["peer.port"] = core.request.get_remote_client_port(ctx),
+            ["peer.service"] = conf.service_name,
+            ["local.ip"] = ctx.var.server_addr,
```
Comment (moonming) [inline lua/apisix/plugins/zipkin.lua]: can we add test cases for that?

File: lua/apisix/plugins/zipkin.lua [hunk 8]
```diff
@@ -131,6 +144,9 @@ function _M.access(conf, ctx)
     opentracing.proxy_span = opentracing.request_span:start_child_span(
             "apisix.proxy", ctx.ACCESS_END_TIME)
 
+    opentracing.proxy_span:set_tag("peer.service", conf.service_name)
```
Comment (membphis) [inline lua/apisix/plugins/zipkin.lua]: do we need a  test case to confirm this?

File: doc/plugins/zipkin-cn.md [hunk 9]
```diff
@@ -118,3 +120,45 @@ $ curl http://127.0.0.1:2379/v2/keys/apisix/routes/1 -X PUT -d value='
 ```
 
 现在就已经移除了 Zipkin 插件了。其他插件的开启和移除也是同样的方法。
+
+
+## 上游服务是Golang的示例代码
+
+```golang
+
```
Comment (membphis) [inline doc/plugins/zipkin-cn.md]: remove this useless line

File: lua/apisix/plugins/zipkin.lua (hunk 7 above)
Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua]: sure I am adding more test cases

File: lua/apisix/plugins/zipkin.lua (hunk 8 above)
Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua]: sure I am adding more test cases

File: doc/plugins/zipkin-cn.md (hunk 9 above)
Comment (lilien1010) [reply doc/plugins/zipkin-cn.md]: sure

File: lua/apisix/plugins/zipkin.lua (hunk 6 above)
Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua]: fixed

File: lua/apisix/plugins/zipkin.lua [hunk 10]
```diff
@@ -29,7 +29,17 @@ local schema = {
...
     properties = {
         endpoint = {type = "string"},
-        sample_ratio = {type = "number", minimum = 0.00001, maximum = 1}
+        sample_ratio = {type = "number", minimum = 0.00001, maximum = 1},
+        service_name = {
+            type = "string",
+            description = "service name for zipkin reporter",
+            default = "APISIX",
+        },
+        server_addr = {
```
Comment (moonming) [inline lua/apisix/plugins/zipkin.lua:38]: need add doc for `server_addr`

Comment (moonming) [inline lua/apisix/plugins/zipkin.lua:38]: need add doc for `server_addr`

Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua:38]: sure

Comment (lilien1010) [reply lua/apisix/plugins/zipkin.lua:38]: done

**Pull request closed without merging**